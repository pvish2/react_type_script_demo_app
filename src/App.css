.App {
  text-align: left;
  width:100%;
  align-items: left
}

.App-logo {
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #09d3ac;
}


.parentDiv{
  padding: 20px;
}

.addDiv{
  margin-top: 20px;
  margin-bottom: 20px;
}
.existingDiv{
  margin-top: 30px;
  margin-bottom: 30px;
}

input[type=number]::-webkit-inner-spin-button, 
input[type=number]::-webkit-outer-spin-button { 
      -webkit-appearance: none; 
      margin: 0; 
}

.button{
  background-color: #B8B8B8	;
  border: none;
  color: white;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 12px;
  margin: 4px 2px;
  cursor: pointer;
  height:35px;
  width: 100px;
}

.errorDiv{
  margin-top: 20px;
  margin-bottom: 20px;
  color:red;
}

.StandardTable {
  border: 1px solid #656565;
  -moz-border-radius: 0px;
  -webkit-border-radius: 0px;
  margin-left: auto;
  margin-right: auto;
  width: 100%;
  margin-bottom: 15px;
  border-spacing: 0;
}
.StandardTable thead {
  border: 1px solid #dddddd;
  border-radius: 0px;
  -moz-border-radius: 0px;
  border-radius: 5px;
  background-color: lightgray;
  text-align: center;
  width : "100%";

}
.StandardTable thead th{
  border: 1px solid #dddddd;
  background: lightgray; 
  text-align: center;
  width : "25%";
  padding: 10px;
}
.StandardTable tbody {
  border: 1px solid #dddddd;
}
.StandardTable tbody tr{
  border: 1px solid #dddddd;
  text-align: center;
  width : "100%";
  padding: 10px;

}
.StandardTable tbody td{
  border: 1px solid #dddddd;
  text-align: center;
  width : "25%";
  padding: 10px;

}

.StandardTable thead th:first-of-type{
  border-top-left-radius: 0px; 
}

.StandardTable thead th:last-of-type{
  border-top-right-radius: 0px; 
}

.label {
  display: block;
  float: left;
  white-space: pre-line;
}

.input{
  margin-top: 5px;
  height:40px;
  width:100%;
  border-color: #B8B8B8	;  
  border-style: solid;
  padding-left: 5px;
}

.inputDiv{
  margin-bottom: 10px;
  margin-left: 0px;
  width: 40%;
}