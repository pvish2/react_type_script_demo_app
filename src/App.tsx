import React, { Component } from "react";
import "./App.css";

import Constants from "./constants/Card";
import Card from "./components/Cards";
import CreditCardForm from "./components/CreditCardForm";

export interface IProps {}

interface Card {
  name: string;
  cardNumber: number;
  limit: number;
}
interface IState {
  cards: Card[];
}

class App extends Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    this.state = {
      cards: []
    };
    this.getAllCreditCardList = this.getAllCreditCardList.bind(this);
  }

  componentDidMount() {
    this.getAllCreditCardList();
  }

  getAllCreditCardList() {
    fetch(Constants.API_URL)
      .then(res => res.json())
      .then(data => {
        console.log(data);
        this.setState({ cards: data });
      })
      .catch(e => console.log(e));
  }

  render() {
    const { cards } = this.state;
    return (
      <div className="App">
        <div className="parentDiv">
          <h2>Credit Card System</h2>
          <h4 className="addDiv"> Add</h4>
          <CreditCardForm refreshCallback={this.getAllCreditCardList} />
          <div></div>
          <h4 className="existingDiv">Existing Cards</h4>
          <Card cards={cards} />
        </div>
      </div>
    );
  }
}

export default App;
