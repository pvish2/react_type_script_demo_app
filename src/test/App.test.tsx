import React from "react";
import App from "../App";
import { shallow } from "enzyme";
import Constants from "../constants/Card";

describe("App", () => {
  let component = "";

  beforeAll(() => {
    component = shallow(<App />);
  });

  it("should render my app", () => {
    expect(component.getElements()).toMatchSnapshot();
  });

  it("fetches data from server when server returns a successful response", done => {
    const mockSuccessResponse = {};
    const mockJsonPromise = Promise.resolve(mockSuccessResponse);
    const mockFetchPromise = Promise.resolve({
      json: () => mockJsonPromise
    });
    jest.spyOn(global, "fetch").mockImplementation(() => mockFetchPromise);
    shallow(<App />);
    expect(global.fetch).toHaveBeenCalledTimes(1);
    expect(global.fetch).toHaveBeenCalledWith(Constants.API_URL);

    process.nextTick(() => {
      expect(component.state()).toEqual({
        cards: []
      });
      global.fetch.mockClear();
      done();
    });
  });
});
