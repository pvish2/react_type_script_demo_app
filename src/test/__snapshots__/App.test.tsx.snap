// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render my app 1`] = `
Array [
  <div
    className="App"
  >
    <div
      className="parentDiv"
    >
      <h2>
        Credit Card System
      </h2>
      <h4
        className="addDiv"
      >
         Add
      </h4>
      <CreditCardForm
        cardNumber="number"
        limit={0}
        name="string"
        refreshCallback={[Function]}
      />
      <div />
      <h4
        className="existingDiv"
      >
        Existing Cards
      </h4>
      <Cards
        cards={Array []}
      />
    </div>
  </div>,
]
`;
